internal/service/url_service.go

```diff
@@ -21,23 +25,72 @@ func NewURLService(repo repository.URLRepository) URLService {
 }
 
 func (s urlService) CreateShortURL(req models.ShortenRequest) (string, error) {
-	//TODO implement me
-	panic("implement me")
+	short := shortener.GeneratorShortURL()
+	originalurl := req.URL
+	timeat := time.Now()
+
+	respons := models.URL{
+		ID:          uuid.NewString(),
+		OriginalURL: originalurl,
+		ShortURL:    short,
+		CustomAlias: req.CustomAlias,
+		CreatedAt:   timeat,  //maybe we will use
+		ExpiredAt:   &timeat, //the time in a different way
+	}
+
+	err := s.repo.SaveURL(respons)
+	if err != nil {
+		return "", fmt.Errorf("create short url, get url err: %s ", err)
+	}
+	return short, nil
 }
 
 func (s urlService) GetOriginalURL(shortURL string) (string, error) {
-	//TODO implement me
-	panic("implement me")
+
+	originalUrl, err := s.repo.GetURL(shortURL)
+	if err != nil {
+		return "", fmt.Errorf("short url not found %s", shortURL)
```

Comment:
why in error send "shortURL" and not "err"

Reply:
and you need fix format of all errors you return, like previously: fmt.Errorf("create short url, get url err: %s ", err)